### PR TITLE
add sds to snapshots such that secrets can be dynamically configured

### PR DESCRIPTION
**Pull request by stevesloka**
Updates #57 by adding SDS to snapshots so that secrets can be dynamically configured. 

Signed-off-by: Steve Sloka <slokas@vmware.com>

### CODE REVIEW

**Comment by tessykt** [comment]
Hello!

Do you know when this code is expected to be added into the master branch?
I was hoping to start using it :)

Thank you in advance for your time!

**Comment by kyessenov** [comment]
It should be possible to use SDS as-is. This PR improves upon the interfaces, but you can access the state directly without the accessors.

File: go.mod [hunk 1]
```diff
@@ -6,10 +6,11 @@ require (
 	github.com/census-instrumentation/opencensus-proto v0.2.1
 	github.com/cncf/udpa/go v0.0.0-20200313221541-5f7e5dd04533
 	github.com/envoyproxy/protoc-gen-validate v0.1.0
-	github.com/golang/protobuf v1.3.2
-	github.com/google/go-cmp v0.2.0
+	github.com/golang/protobuf v1.4.1
```
Comment (kyessenov) [inline go.mod:9]: Can we leave update to another PR?

Comment (stevesloka) [reply go.mod]: @kyessenov I removed that update from this PR.
